indico/modules/events/abstracts/util.py

```diff
@@ -0,0 +1,138 @@
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from collections import OrderedDict
+
+from indico.core.db import db
+from indico.modules.events.abstracts.models.abstracts import Abstract
+from indico.modules.events.util import ListGeneratorBase
+from indico.util.i18n import _
+from indico.util.string import to_unicode
+from indico.web.flask.templating import get_template_module
+
+
+class AbstractListGenerator(ListGeneratorBase):
+    """Listing and filtering actions in the abstract list."""
+
+    endpoint = '.manage_abstract_list'
+    list_link_type = 'abstract'
+
+    def __init__(self, event):
+        super(AbstractListGenerator, self).__init__(event)
+        self.default_list_config = {
+            'items': ('proposed_type', 'accepted_type', 'proposed_track', 'accepted_track'),
+            'filters': {'fields': {}, 'items': {}}
+        }
+        # TODO: session_empty = {None: 'No session'}
+        track_empty = {None: 'No track'}
+        type_empty = {None: 'No type'}
+        track_choices = {unicode(t.id): to_unicode(t.getTitle()) for t in self.list_event.as_legacy.getTrackList()}
+        type_choices = {unicode(t.id): t.name for t in self.list_event.contribution_types}
+        self.static_items = OrderedDict([
+            # TODO: ('session', {'title': _('Session'),
+            #                    'filter_choices': OrderedDict(session_empty.items() + session_choices.items())}),
+            ('accepted_track', {'title': _('Accepted track'),
+                                'filter_choices': OrderedDict(track_empty.items() + track_choices.items())}),
+            ('proposed_track', {'title': _('Proposed track'),
+                                'filter_choices': OrderedDict(track_empty.items() + track_choices.items())}),
+            ('accepted_type', {'title': _('Accepted type'),
+                               'filter_choices': OrderedDict(type_empty.items() + type_choices.items())}),
+            ('proposed_type', {'title': _('Proposed type'),
+                               'filter_choices': OrderedDict(type_empty.items() + type_choices.items())})
+        ])
+        self.list_config = self._get_config()
+
+    def _get_static_columns(self, ids):
+        """
+        Retrieve information needed for the header of the static columns.
```

Comment:
copying the docstring when subclassing is not really important. it just adds a rather high chance of people updating it only here or only in the base class but not both of them